subtests/docker_cli/systemd/systemd.py

```diff
@@ -0,0 +1,81 @@
+"""
+Summary
+-------
+Testing basic container as systemd service
+
+Operational Summary
+---------------------
+This is to test container as systemd service using a predefined unit file
+(p4321.service) and a script p4321-server.py. An image is built with this
+script added. Systemd starts a container using this image & the container
+writes current time to PORT 4321. From host, the socket is read and the value
+is checked to see if it is a digit. Finally, container is stopped through
+systemd and system is cleaned up.
+
+"""
+
+import shutil
+import time
+import socket
+import errno
+from os.path import join
+from os import unlink
+from autotest.client import utils
+from autotest.client.shared.utils import is_port_free
+from dockertest.subtest import Subtest
+from dockertest.dockercmd import DockerCmd
+
+
+class systemd(Subtest):
+
+    def initialize(self):
+        super(systemd, self).initialize()
+        self.stuff['systemd_touchstone'] = False
+        sysd_unitf = join(self.bindir, 'p4321.service')
+        shutil.copyfile(sysd_unitf, self.config['sysd_unitf_dest'])
+        self.build_image()
+
+    def build_image(self):
+        dkrcmd = DockerCmd(self, 'build', ['--force-rm -t',
+                                           self.config['name'], self.bindir])
+        dkrcmd.execute()
```

Comment:
I think it might be wise to wrap this in `mustpass()`, i.e.

    from dockertest.output import mustpass
    ...
    mustpass(dkrcmd.execute())